pyro/distributions/transforms/discrete_cosine.py

```diff
@@ -73,3 +73,8 @@ def _inverse(self, y):
 
     def log_abs_det_jacobian(self, x, y):
         return x.new_zeros((1,) * self.event_dim)
+
+    def with_cache(self, cache_size=1):
+        if self._cache_size == cache_size:
+            return self
+        return DiscreteCosineTransform(-self.event_dim, self.smooth, cache_size)
```

Comment:
Please specify `cache_size=cache_size` as a kwarg. In particular, I plan on adding new kwargs to `DiscreteCosineTransform` and will start using the `*,` separator to force kwarg specification.

Reply:
I've made this change